sos/report/plugins/hyperv.py

```diff
@@ -0,0 +1,30 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+
+
+class Hyperv(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+    """Hyper-V client information"""
+
+    plugin_name = "hyperv"
+    files = ('/sys/bus/vmbus/', '/boot/config-*')
+
+    def setup(self):
+
+        self.add_copy_spec([
+            "/sys/bus/vmbus/drivers/",
+            # copy devices/*/* instead of devices/ to follow link files
+            "/sys/bus/vmbus/devices/*/*",
+            "/boot/config-*"
```

Comment:
I'm not sure that the `config-` files should be collected, but in the event they should be, the `boot` plugin would be the place for this collection.

Reply:
Those files are the build options for the installed kernels, so if they belong anywhere it's with kernel data rather than boot data (they are installed to `/boot` so that they are available for inspection with the kernel and initramfs images they belong to but from a support stance they are closer to the kernel).

Reply:
Thank you for the tip! Is there a reason this isn't currently being collected? It seems to contain a lot of information about the kernel configuration.

Reply:
Mostly that they are static files: they are installed by the kernel RPM package (on Red Hat), and they represent the options used to build the kernel. They cannot be changed without rebuilding the kernel so as long as you know the kernel version you can always get the config-* content without needing to collect it from the machine. If the kernel has been rebuilt then we can detect that in other ways and the content of the file is likely not up to date anyway.

Reply:
Hey Bryn, I looked into this a little bit. From the context of kernel developers debugging a VM issue, having the current kernel configurations is very useful. You are right that `/boot/config-*` may be out of date, may have many old configs (from the kernel package getting updated), and could be obtained manually if you know the kernel package installed. However, I would posit that it would be very useful for sosreport to opportunistically collect the _current_ kernel's build configs, which _can be_ available at `/proc/config.gz`. This file being exposed by the kernel is dependent on the kernel having been built with the config module enabled, but may we propose collecting it when it's available in the sosreport kernel plugin?

Reply:
Getting the compiled-in configuration is different, because it's strongly bound to the kernel image - but - the last time I looked it was not enabled on many distros. At least historically Red Hat (inc. Fedora/RHEL/CentOS), Debian, and Ubuntu don't enable the CONFIG_IKCONFIG_PROC option that would cause this file to be available.

These distros all provide the `/boot/config-*` files as an alternative.